### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [17]:
import numpy as np
import math
a=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150 ])
n=len(a)
mean=sum(a)/n
deviations = [(x - mean) ** 2 for x in a]
variance = sum(deviations) / n
var = sum(deviations) / (n-1)
std_dev = math.sqrt(variance)
print(f'Среднее арифметическое: {mean}')
print(f'Среднее квадратичное отклонение: {std_dev}')
print(f'Смещённая оценка дисперсии: {variance}')
print(f'Несмещённая оценка дисперсии: {var}')

Среднее арифметическое: 65.3
Среднее квадратичное отклонение: 30.823854398825596
Смещённая оценка дисперсии: 950.11
Несмещённая оценка дисперсии: 1000.1157894736842


Ответ: Среднее арифметическое: 65.3;
Среднее квадратичное отклонение: 30.823854398825596;
Смещённая оценка дисперсии: 950.11;
Несмещённая оценка дисперсии: 1000.1157894736842.


_______________________________________________________________________________________________________________________________

### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [21]:
from math import factorial

In [68]:
def combinations(n,k):
    return int(factorial(n)/(factorial(k)*factorial(n-k)))

In [47]:
# Первый вариант, когда из первой корзины вытаскиваем все небелый, а из второй 3 белых и один не белый
p1=combinations(3,2)/combinations(8,2) * (combinations(5,3)*combinations(7,1)/combinations(12,4))
# Второй вариант, из первой корзины берём один белый мяч, а из второй 2 белых мяча
p2=(combinations(5,1)*combinations(3,1)/combinations(8,2))*(combinations(5,2)*combinations(7,2)/combinations(12,4))
# Третий вариант, из первой корзины берём 2 белых мяча, из второй один белый
p3=(combinations(5,2)/combinations(8,2))*(combinations(5,1)*combinations(7,3)/combinations(12,4))
print(f'вероятность того, что 3 мяча белые: {p1+p2+p3}')

вероятность того, что 3 мяча белые: 0.3686868686868687


Ответ: вероятность того, что 3 мяча белые: 0.3686868686868687

_______________________________________________________________________________________________________________________________

### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

In [59]:
# Воспользуемся формулой Байеса

Pa=1/3*0.9+1/3*0.8+1/3*0.6
print(f'вероятность попадания в мишень: {Pa}')
print(f'вероятность того, что выстрел произведен первым спортсменом: {1/3*0.9/Pa}')
print(f'вероятность того, что выстрел произведен первым спортсменом: {1/3*0.8/Pa}')
print(f'вероятность того, что выстрел произведен первым спортсменом: {1/3*0.6/Pa}')

вероятность попадания в мишень: 0.7666666666666666
вероятность того, что выстрел произведен первым спортсменом: 0.391304347826087
вероятность того, что выстрел произведен первым спортсменом: 0.3478260869565218
вероятность того, что выстрел произведен первым спортсменом: 0.2608695652173913


Ответ: вероятность того, что выстрел произведен первым спортсменом: 0.391304347826087;
вероятность того, что выстрел произведен первым спортсменом: 0.3478260869565218;
вероятность того, что выстрел произведен первым спортсменом: 0.2608695652173913.

_______________________________________________________________________________________________________________________________

### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [62]:
# Воспользуемся формулой Байеса
Pa=1/4*0.8+1/4*0.7+2/4*0.9
print(f'вероятность сдачи сессии: {Pa}')
print(f'вероятность того, что студент учился на факультете А: {1/4*0.8/Pa}')
print(f'вероятность того,  что студент учился на факультете В: {1/4*0.7/Pa}')
print(f'вероятность того, что студент учился на факультете С: {2/4*0.9/Pa}')

вероятность сдачи сессии: 0.825
вероятность того, что студент учился на факультете А: 0.24242424242424246
вероятность того,  что студент учился на факультете В: 0.21212121212121213
вероятность того, что студент учился на факультете С: 0.5454545454545455


Ответ: 
вероятность того, что студент учился на факультете А: 0.24242424242424246;

вероятность того,  что студент учился на факультете В: 0.21212121212121213;

вероятность того, что студент учился на факультете С: 0.5454545454545455.

_______________________________________________________________________________________________________________________________

### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [73]:
p1=0.1
p2=0.2
p3=0.25
print(f'вероятность того, что в первый месяц выйдут из строя все детали: {p1*p2*p3}')
print(f'вероятность того, что в первый месяц выйдут из строя только две детали: {p1*p2*+p2*p3+p1*p3}')
print(f'вероятность того, что в первый месяц выйдет из строя хотя бы одна деталь: {1-(0.9*0.8*0.75)}')
print(f'вероятность того, что в первый месяц выйдут из строя от одной до двух деталей: {(1-(0.9*0.8*0.75))-p1*p2*p3}')


вероятность того, что в первый месяц выйдут из строя все детали: 0.005000000000000001
вероятность того, что в первый месяц выйдут из строя только две детали: 0.026000000000000002
вероятность того, что в первый месяц выйдет из строя хотя бы одна деталь: 0.45999999999999996
вероятность того, что в первый месяц выйдут из строя от одной до двух деталей: 0.45499999999999996


Ответ: 
вероятность того, что в первый месяц выйдут из строя все детали: 0.005000000000000001

вероятность того, что в первый месяц выйдут из строя только две детали: 0.026000000000000002

вероятность того, что в первый месяц выйдет из строя хотя бы одна деталь: 0.45999999999999996

вероятность того, что в первый месяц выйдут из строя от одной до двух деталей: 0.45499999999999996
    